### ISSUE TITLE
Editor timeout

### ISSUE THREAD

**Issue by matejak** [exo-addons/onlyoffice#12]
I use the addon with ExoPlatform 4.4 with this addon installed. The integration into the interface woks, I am able to click the "Edit in OnlyOffice" button. However, I keep getting the "Error creating editor: ONLYOFFICE script load timeout. Ensure Document Server is running and accessible".

First of all, I have noticed that I have forgot to configure the addon in the `exo.properties`, but the situation did not change even after adding the configuration and restarting the server. I find this somehow suspicious, so I don't rule out that the configuration is somehow still broken. But it may be related to #11 as well.

Are there any log files that I could look into to find out what is the addon connecting to and what it expects, but doesn't get?

**Comment by pnedonosko** [comment]
Hi Matej, it looks as a configuration issue. Ensure that document server URL you configured is the same as an one publicly accessible (by other users in browsers). The add-on will try load Onlyoffice editor scripts from it, and your error tells it failed. 
For more details you need look in the browser console. Server log is less useful for this kind of issue.
Append you browser console here if need more guidance.

**Comment by matejak** [comment]
Thank you for your fast reply,
I can see that I have misunderstood the instructions. However, it still doesn't work for me. After I set up a server (let's say it's `office.example.com`, but it is a valid DNS record), I get these messages in the browser console:
```
Blocked loading mixed active content “http://office.example.com/2017-11-10-16-41/web-apps/apps/api/documents/api.js?_=1512429736160”
Loading failed for the <script> with source “https://office.example.com/OfficeWeb/apps/api/documents/api.js?_=1512429736160”.
```
and there is still the "timeout" error.

The `exo.properties` contains
```
onlyoffice.documentserver.host=office.example.com:443
onlyoffice.documentserver.schema=https
```
The strange thing is the `http` in the first line of the console report, and I am totally confused by the second line, because when I enter that URL in the browser, I get the javascript code without issues. (And even the first link works, because there is a http -> https redirect with valid certificate). Any ideas what might be wrong?

**Comment by matejak** [comment]
And I should probably add that request for
```
https://office.example.com/OfficeWeb/apps/api/documents/api.js?_=1512431701556
```
ends with the 302 status, redirect points to
```
http://office.example.com/2017-11-10-16-41/web-apps/apps/api/documents/api.js?_=1512431701556
```
but it doesn't seem to be followed (judging by observation of the Firefox Network Monitor, but I am not a skilled networking debugger). Needless to say, there is another 301 redirect (http -> https) waiting for the request, and I don't see that at all in the monitor.

**Comment by pnedonosko** [comment]
It looks bad: Onlyoffice's script is on HTTPS, but it redirects it to HTTP (not secure) and then the browser fails to load it and it is right. May be try use another version of Document Server (another docker image version). It is something new and may be need search or ask in Onlyoffice forum also.

**Comment by matejak** [comment]
You are completely right! However, I have managed to solve the http/https issue by following this issue: https://github.com/ONLYOFFICE/Docker-CommunityServer/issues/10 namely by requesting `proxy_redirect http://office.example.org/ https://office.example.org/;`

**Comment by pnedonosko** [comment]
Matej, thank you for sharing your findings. Can you put here a piece of that browser console about {http} and "Unknown error"?

**Comment by pnedonosko** [comment]
@matejak have you solved this issue?

**Comment by matejak** [comment]
Hello, sorry for the delay. I have followed the last instruction from https://github.com/ONLYOFFICE/Docker-CommunityServer/issues/10 just before the issue was closed, and I have succeeded in making it work. The trick with `production-linux.json` is a modification of the OnlyOffice setup (rather than the addon setup).
People that run Exoplatform and Onlyoffice behind a proxy server are likely to encounter the same issues that I had encountered, others will be probably fine.
